lib/elixir/lib/code.ex

```diff
@@ -324,6 +324,39 @@ defmodule Code do
     unrequire_files(files)
   end
 
+  @doc ~S"""
+  Returns the list of lines in the given string, preserving their line endings.
+
+  Only the line endings recognized by the Elixir compiler are
+  considered, namely `\r\n` and `\n`. If you would like the retrieve
+  lines without their line endings, use `String.split(string, ["\r\n", "\n"])`.
```

Comment:
It feels like the disclaimer about ending will narrow the use case for this quite a bit?
Perhaps we could add a `trim: true` option later to remove the ending, in which case it would make it more generally useful.
Which kind of use cases do we have in mind for this?

Reply:
The use case is to fetch a range from a source file but preserving its original source code. I can add `trim: true` but `trim` typically means trimming more (spaces, tabs, etc). It is also something folks can do by doing another pass on the data.